src/larvaworld/lib/model/modules/sensor.py

```diff
@@ -212,3 +213,26 @@ def cool_sensor_input(self):
     @property
     def cool_sensor_perception(self):
         return self.dX['cool']
+
+
+class BrianOlfactor(Olfactor):
+    def __init__(self, response_key='OSN_rate', server_host='localhost', server_port=5795, remote_dt=100, remote_warmup=0, **kwargs):
+        super().__init__(**kwargs)
+        self.brianInterface = RemoteBrianModelInterface(server_host, server_port, remote_dt)
+        self.brian_warmup = remote_warmup
+        self.response_key = response_key
+        self.remote_dt = remote_dt
+
+
+    def update(self):
+        agent_id = self.brain.agent.unique_id # TODO: is this always present ?
+        msg_kws = {
+            # Default :
+            'odor_id': 0, # TODO: can we get this info from somewhere ?
```

Comment:
not required not - but can I get an odor ID from somewhere which this sensor is attached/tuned to?